stdlib/asm/collections/mmr.masm

```diff
@@ -104,3 +106,76 @@ export.get
      # stack: [leaf, ...]
   end
 end
+
+#! Load the MMR peak data based on its hash.
+#!
+#! Input: [HASH, mmr_ptr, ...]
+#! Output: [...]
+#!
+#! Where:
+#!  - HASH: is the MMR peak hash, the hash is expected to be padded to an even
+#!    length and to have a minimum size of 16 elements
+#!  - mmt_ptr: the memory location where the MMR data will be written to,
+#!    starting with the MMR forest (its total leaves count) followed by its peaks
+#!
+#! Cycles: 159 + 9 * extra_peak_pair cycles
```

Comment:
Note: Using a fixed hasher input would use fewer cycles.

Fixed 32 (69 cycles)

```
    padw padw padw                      # prepare hasher state, no padding is required (16 cycles)
    adv_pipe adv_pipe adv_pipe adv_pipe # words 0..8 (8 cycles)
    adv_pipe adv_pipe adv_pipe adv_pipe # words 8..16 (8 cycles)
    adv_pipe adv_pipe adv_pipe adv_pipe # words 16..24 (8 cycles)
    adv_pipe adv_pipe adv_pipe adv_pipe # words 24..32 (8 cycles)
    dropw swapw dropw                   # hash result (9 cycles)
    movup.4 drop                        # drop write_ptr (2 cycles)
    assert_eqw                          # check commitment (10 cycles)
```

Fixed 64 (101 cycles)

```
    padw padw padw                      # prepare hasher state, no padding is required (16 cycles)
    adv_pipe adv_pipe adv_pipe adv_pipe # words 0..8 (8 cycles)
    adv_pipe adv_pipe adv_pipe adv_pipe # words 8..16 (8 cycles)
    adv_pipe adv_pipe adv_pipe adv_pipe # words 16..24 (8 cycles)
    adv_pipe adv_pipe adv_pipe adv_pipe # words 24..32 (8 cycles)
    adv_pipe adv_pipe adv_pipe adv_pipe # words 32..40 (8 cycles)
    adv_pipe adv_pipe adv_pipe adv_pipe # words 40..48 (8 cycles)
    adv_pipe adv_pipe adv_pipe adv_pipe # words 48..56 (8 cycles)
    adv_pipe adv_pipe adv_pipe adv_pipe # words 56..64 (8 cycles)
    dropw swapw dropw                   # hash result (9 cycles)
    movup.4 drop                        # drop write_ptr (2 cycles)
    assert_eqw                          # check commitment (10 cycles)
```

Reply:
I'm actually fine with the current cycle count. It is a balance between "in-VM" and "out-of-VM" performance and if we go with fixed size 64 or even 32, I think we give up quite a bit on the "out-of-VM" front.